client/client/app/components/ngbDataSets/internal/ngbDataSetMetadata/ngbDataSetMetadata.controller.js

```diff
@@ -60,8 +60,9 @@ export default class ngbDataSetMetadataController extends BaseController {
     }
     get existedKeys() { 
         return this.formData.map(pair => pair[0]).reduce((r, key) => {
+            const caseInsensitiveKey = key ? key.toLocaleLowerCase() : null;
             if (key) {
```

Comment:
Though this will work, it's error-prone: if somebody change the initiation of `caseInsensitiveKey` (e.g., `caseInsensitiveKey = key && SOME_FALSE_CRITERIA ? key.toLocaleLowerCase() : null`), your code will fail because `if (key)` will pass, but `r[caseInsensitiveKey]` will fail.
Change this to `if (caseInsensitiveKey)` please.
Also, change `toLocaleLowerCase()` to `toLowerCase()` - again, in mostly cases this is the same; but there are few (in some locales) cases when results will differ. We need to turn this string to lower cased format in ALL cases